packages/application-to-register-webapp/src/routes/land/add-legal-agreement-parties.js

```diff
@@ -0,0 +1,177 @@
+import constants from '../../utils/constants.js'
+
+const START_ID = '#organisation'
+let referredFrom
```

Comment:
I do not think module level variables are safe for concurrent processing. While this approach will work for single client usage such as local development and unit testing, concurrent processing runs the risk of a session using referrer values from another session.

@teddmason  is working on an alternative that stores referrer data in the Redis cache so that it is scoped to a particular session. I suggest liaising with @teddmason to introduce the functionality to this pull request. Associated unit tests might need refactoring accordingly.

Reply:
I agree. I will liaise with Ted on this.